lib/kernel/SPIR/generate_spir_layer.py

```diff
@@ -0,0 +1,306 @@
+#!/usr/bin/python3
+#
+# A script to generate functions (SPIR-mangled with SPIR AS) that will call
```

Comment:
Like we discussed, I think this is good enough for now. Later we might consider integrating this functionality to linker.cpp so it maps the SPIR-V AS0 ones on the fly to the target library ones.

Reply:
Yes. It should be also fairly easy to extend it to support TCE and CUDA, both use SPIR AS numbers.